### PR TITLE
camlidl.1.05: disable fatal warning for ocaml version >= 4.03

### PR DESCRIPTION
**Pull request by nberth**
Warnings about duplicate module names are fatal by default since 4.03
due to dangerous semantics for double-linking of modules involving
extensible variant types; yet I do not think this is much of a problem
here as long as the dupicate module in camlidl (Array) does not
involve this feature.

### CODE REVIEW

**Comment by yallop** [comment]
Can this be fixed with a patch instead?

See [packages/sqlite3/sqlite3.1.6.3/files/opam.patch] (https://github.com/ocaml/opam-repository/blob/9f47c1213841a2e9ba08a48ee14af7a74605cfb0/packages/sqlite3/sqlite3.1.6.3/files/opam.patch) (and [packages/sqlite3/sqlite3.1.6.3/opam](https://github.com/ocaml/opam-repository/blob/master/packages/sqlite3/sqlite3.1.6.3/opam#L17)) for an example.

File: packages/camlidl/camlidl.1.05/opam [hunk 1]
```diff
@@ -1,11 +1,12 @@
-opam-version: "1"
+opam-version: "1.2"
 maintainer: "contact@ocamlpro.com"
 authors: ["Xavier Leroy"]
 homepage: "http://caml.inria.fr/pub/old_caml_site/camlidl/"
 license: "QPL-1.0 and LGPL-2 with exceptions"
 build: [
   ["mv" "config/Makefile.unix" "config/Makefile"] {(os = "darwin") | ((os = "linux") | ((os = "freebsd") | ((os = "bsd") | ((os = "unix") | (os = "cygwin")))))}
   ["mv" "config/Makefile.win32" "config/Makefile"] {os = "win32"}
+  ["sh" "-c" "echo 'OCAMLC=ocamlc -g -warn-error -31' >> config/Makefile" ] { ocaml-version >= "4.03" }
```
Comment (dsheets) [inline packages/camlidl/camlidl.1.05/opam]: Please use `-exc`  when invoking `sh` unless you use @yallop's suggestion of using a patch file, instead.

**Comment by nberth** [comment]
lint complains about the maintainer field (it has the old default value "contact@ocamlpro…"). Yet this package does not have an actual maintainer; what should one put in this field instead?

**Comment by dsheets** [comment]
@nberth The maintainer field tracks the maintainer of the opam package metadata in this repository and not the maintainer of the upstream package (though, those are often the same person). There is no real responsibility to be listed as the maintainer of a package but we usually attempt to notify the maintainer via GitHub @ mention when the package metadata is updated. In the future, we may have automation to send the maintainer an email notification when the package is updated.

Given that you appear be a user of the package and are willing to submit pull requests to the package metadata, I think you should put yourself in the maintainer field. :-)

**Comment by nberth** [comment]
@dsheets OK I already maintain a few OPAM packages that depend on that one (and I need to push them once this one is mainstream), so I'll put myself as maintainer.

**Comment by yallop** [comment]
Thanks!

**Pull request merged**